core/connection/manager.go

```diff
@@ -130,22 +128,19 @@ func (manager *connectionManager) Connect(consumerID identity.Identity, proposal
 }
 
 func (manager *connectionManager) startConnection(consumerID identity.Identity, proposal market.ServiceProposal, params ConnectParams) (err error) {
-	manager.mutex.Lock()
 	cancelCtx := manager.cleanConnection
-	manager.mutex.Unlock()
 
 	var cancel []func()
 	defer func() {
 		manager.cleanConnection = func() {
-			manager.status = statusDisconnecting()
 			cancelCtx()
 			for i := range cancel { // Cancelling in a reverse order to keep correct workflow.
 				cancel[len(cancel)-i-1]()
 			}
 		}
 		if err != nil {
 			log.Info(managerLogPrefix, "Cancelling connection initiation")
-			defer manager.cleanConnection()
+			_ = manager.Disconnect()
```

Comment:
Why not to log an error?

Reply:
The only error we're possibly returning from disconnect is the `ErrNoConnection`, which is going to flood the logs needlessly, since these disconnects will get triggered by multiple goroutines(possibly).